### PR TITLE
SNOW-1296603: Add daily precommit test

### PR DESCRIPTION
**Pull request by sfc-gh-jdu**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #NNNN

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Add daily precommit tests. An example here https://github.com/snowflakedb/snowpark-python/actions/runs/8637285804

### CODE REVIEW

File: .github/workflows/daily_precommit.yml [hunk 1]
```diff
@@ -0,0 +1,528 @@
...
+
+  test-enable-cte-optimization:
+    name: Test Enable CTE Optimization py-${{ matrix.os.download_name }}-${{ matrix.python-version }}-${{ matrix.cloud-provider }}
+    needs: build
+    runs-on: ${{ matrix.os.image_name }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - image_name: macos-latest
```
Comment (sfc-gh-yzou) [inline .github/workflows/daily_precommit.yml:392]: @sfc-gh-jdu for this test, do you need to run it all on macos? if not, maybe run it on the larger machien

File: .github/workflows/daily_precommit.yml [hunk 2]
```diff
@@ -0,0 +1,528 @@
...
+
+  test-no-pandas:
+    name: Test when pandas is not installed py-${{ matrix.os.download_name }}-${{ matrix.python-version }}-${{ matrix.cloud-provider }}
+    needs: build
+    runs-on: ${{ matrix.os.image_name }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - image_name: macos-latest
```
Comment (sfc-gh-yzou) [inline .github/workflows/daily_precommit.yml:271]: maybe use a the larger machine for this one also?

File: .github/workflows/daily_precommit.yml [hunk 3]
```diff
@@ -0,0 +1,528 @@
...
+  test-disable-sql-simplifier: # Will be removed after sql simplifier is stable and no option to opt out.
+    name: Test Disable SQL Simplifier py-${{ matrix.os.download_name }}-${{ matrix.python-version }}-${{ matrix.cloud-provider }}
+    needs: build
+    runs-on: ${{ matrix.os.image_name }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - image_name: macos-latest
+            download_name: macos  # it includes doctest
```
Comment (sfc-gh-yzou) [inline .github/workflows/daily_precommit.yml:204]: maybe use larger machine for this one also

File: .github/workflows/daily_precommit.yml (hunk 1 above)
Comment (sfc-gh-jdu) [reply .github/workflows/daily_precommit.yml:392]: per our offline discussion, macos is required to run doctests, so let's keep it using mac for now

**Review by sfc-gh-yuwang — APPROVED** [review]

**Review by sfc-gh-aling — APPROVED** [review]

**Pull request merged**